packages/unlock-js/src/Unlock/v11/createLock.js

```diff
@@ -0,0 +1,111 @@
+import { ethers } from 'ethers'
+import abis from '../../abis'
+import ethersUtils from '../../utils'
+import { ETHERS_MAX_UINT, UNLIMITED_KEYS_COUNT, ZERO } from '../../constants'
+import { getErc20Decimals } from '../../erc20'
+
+/**
+ * Returns the key price in its currency, rather than its decimal representation (Ether vs. Wei for example)
+ * @param {*} currencyContractAddress
+ * @param {*} lock
+ * @param {*} provider
+ */
+async function _getKeyPrice(lock, provider) {
+  const currencyContractAddress = lock.currencyContractAddress || ZERO
+
+  if (currencyContractAddress !== ZERO) {
+    // We need to get the decimal value
+    const erc20Decimals = await getErc20Decimals(
+      currencyContractAddress,
+      provider
+    )
+    return ethersUtils.toDecimal(lock.keyPrice, erc20Decimals)
+  }
+  return ethersUtils.toWei(lock.keyPrice, 'ether')
+}
+
+/**
+ * Creates a lock at a specific version
+ * @param {PropTypes.lock} lock
+ * @param {number} _lockVersion the version of the lock to create
+ * @param {function} callback invoked with the transaction hash
+ */
+export default async function (lock, lockVersionOrCallback, callback) {
```

Comment:
Let's actually pass the version in the `lock` object for more simplicity?

Reply:
to add the lock number into the lock object will make thing less explicit, also make overloads more cumbersome. Let's keep it here as the implementation, and I will add two different overloads in the exports (with and without lock numbers) to make things clearer.